
syntax = "proto3";

service MessagingService {
  rpc Connect (ConnectRequest) returns (stream Message);
  rpc SendMessage (SendMessageRequest) returns (Empty);
}

message ConnectRequest {
  string sender_name = 1;
}

message SendMessageRequest {
  string sender = 1;
  string recipient = 2;
  string text = 3;
}

message Message {
  string sender = 1;
  string text = 2;
}

message Empty {}
